internal/event/targetlist.go

```diff
@@ -163,46 +183,69 @@ func (list *TargetList) TargetMap() map[TargetID]Target {
 }
 
 // Send - sends events to targets identified by target IDs.
-func (list *TargetList) Send(event Event, targetIDset TargetIDSet, resCh chan<- TargetIDResult, synchronous bool) {
-	if atomic.LoadInt64(&list.currentSendCalls) > maxConcurrentTargetSendCalls {
-		atomic.AddInt64(&list.eventsSkipped, 1)
-		err := fmt.Errorf("concurrent target notifications exceeded %d", maxConcurrentTargetSendCalls)
-		for id := range targetIDset {
-			resCh <- TargetIDResult{ID: id, Err: err}
-		}
-		return
+func (list *TargetList) Send(event Event, targetIDset TargetIDSet, resCh chan<- TargetIDResult, sync bool) {
+	if sync {
+		list.sendSync(event, targetIDset, resCh)
+	} else {
+		list.sendAsync(event, targetIDset, resCh)
 	}
-	if synchronous {
-		list.send(event, targetIDset, resCh)
+}
+
+func (list *TargetList) sendResult(resCh chan<- TargetIDResult, id TargetID, err error) {
+	select {
+	case resCh <- TargetIDResult{
+		ID:  id,
+		Err: err,
+	}:
+	case <-list.ctx.Done():
 		return
 	}
-	go list.send(event, targetIDset, resCh)
 }
 
-func (list *TargetList) send(event Event, targetIDset TargetIDSet, resCh chan<- TargetIDResult) {
+func (list *TargetList) sendSync(event Event, targetIDset TargetIDSet, resCh chan<- TargetIDResult) {
 	var wg sync.WaitGroup
 	for id := range targetIDset {
-		list.RLock()
-		target, ok := list.targets[id]
-		list.RUnlock()
-		if ok {
-			wg.Add(1)
-			go func(id TargetID, target Target) {
-				atomic.AddInt64(&list.currentSendCalls, 1)
-				defer atomic.AddInt64(&list.currentSendCalls, -1)
-				defer wg.Done()
-				tgtRes := TargetIDResult{ID: id}
-				if err := target.Save(event); err != nil {
-					tgtRes.Err = err
-				}
-				resCh <- tgtRes
-			}(id, target)
-		} else {
-			resCh <- TargetIDResult{ID: id}
+		target, ok := list.get(id)
+		if !ok {
+			list.sendResult(resCh, id, nil)
```

Comment:
can we log this error? 

```suggestion
			list.sendResult(resCh, id, fmt.Errorf("target id %v not found", id))
```

or just continue instead by removing this line.

Reply:
it is to satisfy a test, fixed the test instead.